eth/vm/stack.py

```diff
@@ -22,69 +23,213 @@
-                if isinstance(value, int):
-                    yield value
+                raise _busted_type(item_type, popped)
+
+    def pop1_int(self) -> int:
+        """
+        Pop and return an integer from the stack.
+
+        Raise `eth.exceptions.InsufficientStack` if the stack was empty.
+
+        Note: This function is optimized for speed over readability.
+        """
+        if not self.values:
+            raise InsufficientStack(
+                "Wanted 1 stack item, only had %d",
+                len(self.values),
+            )
+        else:
+            item_type, popped = self._pop_internal()
+            if item_type is int:
+                return popped  # type: ignore
+            elif item_type is bytes:
+                return big_endian_to_int(popped)  # type: ignore
+            else:
+                raise _busted_type(item_type, popped)
+
+    def pop1_any(self) -> Union[int, bytes]:
+        """
+        Pop and return an element from the stack.
+        The type of each element will be int or bytes, depending on whether it was
+        pushed with push_bytes or push_int.
+
+        Raise `eth.exceptions.InsufficientStack` if the stack was empty.
+
+        Note: This function is optimized for speed over readability.
+        """
+        if not self.values:
+            raise InsufficientStack(
+                "Wanted 1 stack item, only had %d",
+                len(self.values),
+            )
+        else:
+            _, popped = self._pop_internal()
+            return popped
+
+    def pop_any(self, num_items: int) -> Tuple[Union[int, bytes], ...]:
+        """
+        Pop and return a tuple of items of length ``num_items`` from the stack.
+        The type of each element will be int or bytes, depending on whether it was
+        pushed with stack_push_bytes or stack_push_int.
+
+        Raise `eth.exceptions.InsufficientStack` if there are not enough items on
+        the stack.
+
+        Items are ordered with the top of the stack as the first item in the tuple.
+
+        Note: This function is optimized for speed over readability.
+        """
+        if num_items > len(self.values):
+            raise InsufficientStack(
+                "Wanted %d stack items, only had %d",
+                num_items,
+                len(self.values),
+            )
+        else:
+            neg_num_items = -1 * num_items
+
+            all_popped = reversed(self.values[neg_num_items:])
+            del self.values[neg_num_items:]
+
+            return tuple(val for _, val in all_popped)
+
+    def pop_ints(self, num_items: int) -> Tuple[int, ...]:
+        """
+        Pop and return a tuple of integers of length ``num_items`` from the stack.
+
+        Raise `eth.exceptions.InsufficientStack` if there are not enough items on
+        the stack.
+
+        Items are ordered with the top of the stack as the first item in the tuple.
+
+        Note: This function is optimized for speed over readability.
```

Comment:
Maybe the function could have more comments in it to explain what is hppening and why.  Also, This type of comment should be very visable like:

```
def thing(...):
    """
    ###################################
    #           !!!!! STOP !!!!!      #
    ###################################
```

Something hard to miss when someone decides to edit a line or two.